modules/market/src/main/java/com/opengamma/strata/market/interpolator/LogNaturalDiscountFactorCurveInterpolator.java

```diff
@@ -0,0 +1,232 @@
+     */
+    public static final String NAME = "LogNaturalCubicDiscountFactor";
+    /**
+     * The interpolator instance.
+     */
+    public static final CurveInterpolator INSTANCE = new LogNaturalDiscountFactorCurveInterpolator();
+    /**
+     * The serialization version id.
+     */
+    private static final long serialVersionUID = 1L;
+    
+    private static final MatrixAlgebra MA = new OGMatrixAlgebra();
+    
+    /**
+     * Restricted constructor.
+     */
+    private LogNaturalDiscountFactorCurveInterpolator() {
+    }
+
+    // resolve instance
+    private Object readResolve() {
+      return INSTANCE;
+    }
+
+    //-------------------------------------------------------------------------
+    @Override
+    public String getName() {
+      return NAME;
+    }
+
+    @Override
+    public BoundCurveInterpolator bind(DoubleArray xValues, DoubleArray yValues) {
+      return new Bound(xValues, yValues);
+    }
+
+    //-----------------------------------------------------------------------
+    @Override
+    public String toString() {
+      return NAME;
+    }
+
+    //-------------------------------------------------------------------------
+    /**
+     * Bound interpolator.
+     */
+    static class Bound extends AbstractBoundCurveInterpolator {
+      private final double[] xValues;
+      private final double[] yValues;
+      private final PiecewisePolynomialResultsWithSensitivity poly;
+      private final DoubleArray knots;
+      private final DoubleMatrix coefMatrix;
+      private final int nKnots;
+      private final int dimensions;
+      private double[] logYValues;
+      
+      
+      Bound(DoubleArray xValues, DoubleArray yValues) {
+        super(xValues, yValues);
+        this.xValues = xValues.toArrayUnsafe();
+        this.yValues = yValues.toArrayUnsafe();
+        this.logYValues = getYLogValues(this.yValues);
+        this.poly = new ClampedPiecewisePolynomialInterpolator(new NaturalSplineInterpolator(), new double[] {0d }, new double[] {0d }).interpolateWithSensitivity(xValues.toArray(), logYValues);
+        this.knots = poly.getKnots();
+        this.coefMatrix = poly.getCoefMatrix();
+        this.nKnots= knots.size();
+        this.dimensions = poly.getDimensions();
+      }
+
+      Bound(Bound base, BoundCurveExtrapolator extrapolatorLeft, BoundCurveExtrapolator extrapolatorRight) {
+        super(base, extrapolatorLeft, extrapolatorRight);
+        this.xValues = base.xValues;
+        this.yValues = base.yValues;
+        this.logYValues = base.logYValues;
+        this.poly = base.poly;
+        this.knots = base.knots;
+        this.coefMatrix = base.coefMatrix;
+        this.nKnots= base.nKnots;
+        this.dimensions = base.dimensions;
+      }
+
+      //-------------------------------------------------------------------------
+      private static DoubleArray evaluate(double xValue, DoubleArray knots, DoubleMatrix coefMatrix, int dimensions, int nKnots) {
```

Comment:
Line length on method args. We format each parameter on a new line once it exceeds the maximum width.

This affects a number of methods